tensorflow/docs_src/extend/adding_an_op.md

```diff
@@ -1427,6 +1464,318 @@ compact in representing input and output shape specifications in tests.  For
 now, see the surrounding comments in those tests to get a sense of the shape
 string specification).
 
+## Putting it all together
+For the sake of ease and completeness, here is a full tutorial of how to implement a custom operator which is intended to be used as an activation function.  This isn't actually a very good activation function; it performs worse than commonly used activation functions like ReLu or Sigmoid, but this example is sufficient to use in a neural network and get it to train to marginal accuracy.  This specific mathematical form for the activation function was chosen only to demonstrate clearly how the gradient needs to be calculated, since ReLu is so simple that you can implement its gradient without really being clear what is happening from a mathematical standpoint
+
+1) Create the following three source files in the [`tensorflow/core/user_ops`][user_ops] directory:
+```c++
```

Comment:
Can you put code in code files and link to them in the md? I think that's better than adding them in the md. That'll just go out of date. 

Once you have done that, we absolutely need a test to make sure this actually works.